test/gtest/ucs/test_conn_match.cc

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#include <common/test.h>
+
+#include <ucs/datastruct/conn_match.h>
+#include <ucs/sys/sys.h>
+
+#include <vector>
+
+class test_conn_match : public ucs::test {
+public:
+    typedef struct {
+        ucs_conn_match_elem_t conn_match;
+        bool                  is_exp;
+        void                  *dest_address;
+        ucs_conn_sn_t         conn_sn;
+    } conn_elem_t;
+
+private:
+    conn_elem_t *retrieve_common(ucs_conn_match_elem_t *conn_match,
+                                 const void *dest_address,
+                                 ucs_conn_sn_t conn_sn, bool is_exp) {
+        if (conn_match == NULL) {
+            return NULL;
+        }
+
+        conn_elem_t *conn_elem = ucs_container_of(conn_match, conn_elem_t,
+                                                  conn_match);
+        EXPECT_EQ(is_exp,  conn_elem->is_exp);
+        EXPECT_EQ(conn_sn, conn_elem->conn_sn);
+        EXPECT_TRUE(!memcmp(dest_address, conn_elem->dest_address,
+                            m_conn_match_ctx.address_length));
+        return conn_elem;
+    }
+
+    void insert_common(const void *dest_address, ucs_conn_sn_t conn_sn,
+                       bool is_exp, conn_elem_t &elem) {
+        elem.is_exp  = is_exp;
```

Comment:
you do not need `dest_address` in this func

Reply:
removed